src/zenml/utils/pydantic_utils.py

```diff
@@ -233,40 +256,225 @@ def validate_function_args(
+                f"of `{raw_data.__class__}` (super class of `{base_class}`) "
+                f"has been passed as raw input. This might lead to unexpected "
+                f"behaviour in case `{base_class}` have features which can not"
+                f"be extracted from an instance of a `{raw_data.__class__}`."
+            )
+            return dict(raw_data)
+
+        elif isinstance(raw_data, str):
+            # If the raw input is a raw string, we can try to use the `json`
+            # module to parse it. The resulting data needs to be a proper
+            # dict for us to pass it to the validation process.
+            try:
+                json_data = json.loads(raw_data)
+
+                if isinstance(json_data, dict):
+                    return json_data
+                else:
+                    raise TypeError("The resulting json data is not a dict!")
+
+            except (TypeError, JSONDecodeError) as e:
+                raise TypeError(
+                    "The raw json input string can not be converted to a "
+                    f"dict: {e}"
+                )
+        else:
+            raise TypeError(
+                "Unsupported type of raw input data for the `python` validation"
+                "mode of the pydantic class. Please consider changing the way "
+                f"you are creating using the `{base_class}` or instead use"
+                f"`{base_class}.model_validate_json()`."
+            )
+
+    elif validation_info.mode == "json":
+        # This is mode is only active if people validate objects using json
+        # input data such as MyClass.model_validate_json()
+        if isinstance(raw_data, dict):
+            return raw_data
+        else:
+            raise TypeError(
+                f"The resulting JSON data {raw_data} is not a dict, therefore"
+                f"can not be used by the validation process."
+            )
+    else:
+        # Unknown validation mode
+        raise ValueError(f"Unknown validation mode. {validation_info.mode}")
+
+
+def before_validator_handler(
+    method: Callable[..., Any],
+) -> Callable[[Any, Any, Any], Any]:
+    """Decorator to handle the raw input data for pydantic model validators.
+
+    Args:
+        method: the class method with the actual validation logic.
+
+    Returns:
+        the validator method
+    """
+
+    def before_validator(
+        cls: Type[BaseModel], data: Any, validation_info: ValidationInfo
+    ) -> Any:
+        """Wrapper method to handle the raw data.
+
+        Args:
+            cls: the class handler
+            data: the raw input data
+            validation_info: the context of the validation.
+
+        Returns:
+            the validated data
+        """
+        data = model_validator_data_handler(
+            raw_data=data, base_class=cls, validation_info=validation_info
+        )
+        return method(cls=cls, data=data)
+
+    return before_validator
+
+
+def check_validators(
```

Comment:
I think this would be clearer as `has_validators`, with the current name I was confused what it actually is validating (validators?, no validators?)

Reply:
Good idea, it's much more fitting. Changed it accordingly.